static/src/javascripts/components/raven-js/raven.js

```diff
@@ -1835,4 +1835,4 @@ if (typeof define === 'function' && define.amd) {
     define('raven', [], function() { return Raven; });
 }
 
-})(this);
+})(window);
```

Comment:
I think your version is much better. I never understood (1) why use `this` instead of the explicit `window` in the first place, it is a vestige of unsafe JavaScript, and (2) why pass the global context as an argument, instead of a proxy object, for dependency injection.

I'm curious though, do you remember what those Windows errors were exactly? Rich came across a similar problem with Sonobi and had to [patch their implementation of `Object.getOwnPropertyNames`](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/commercial/modules/dfp/prepare-sonobi-tag.js#L29). Perhaps we could try that too with Raven?